cli/cmd/devtool/start.go

```diff
@@ -0,0 +1,759 @@
+	cmd.Flags().BoolVar(&refreshDotenv, "refresh-dotenv", true, "Reset local development state")
+	services.addFlags(cmd)
+
+	return cmd
+}
+
+func start(ch *cmdutil.Helper, preset string, verbose, reset, refreshDotenv bool, services *servicesCfg) error {
+	ctx := graceful.WithCancelOnTerminate(context.Background())
+
+	err := errors.Join(
+		checkGoVersion(),
+		checkNodeVersion(ctx),
+		checkDocker(ctx),
+		checkRillRepo(ctx),
+	)
+	if err != nil {
+		return err
+	}
+
+	switch preset {
+	case "cloud":
+		err = cloud{}.start(ctx, ch, verbose, reset, refreshDotenv, services)
+	case "local":
+		err = local{}.start(ctx, verbose, reset, services)
+	default:
+		err = fmt.Errorf("Unknown preset %q", preset)
+	}
+	// If ctx.Err() != nil, we don't return the err because any graceful shutdown will cause sub-commands to return non-zero exit code errors.
+	// In these cases, ignoring the error doesn't really matter since "real" errors are probably also logged to stdout anyway.
+	if err != nil && ctx.Err() == nil {
+		return err
+	}
+	return nil
+}
+
+func checkGoVersion() error {
+	v := version.Must(version.NewVersion(strings.TrimPrefix(runtime.Version(), "go")))
+	min := version.Must(version.NewVersion(minGoVersion))
+	if v.LessThan(min) {
+		return fmt.Errorf("Go version %s or higher is required", minGoVersion)
+	}
+	return nil
+}
+
+func checkNodeVersion(ctx context.Context) error {
+	out, err := newCmd(ctx, "node", "--version").Output()
+	if err != nil {
+		return fmt.Errorf("error executing the 'node --version' command: %w", err)
+	}
+
+	v := version.Must(version.NewVersion(strings.TrimSpace(string(out))))
+	min := version.Must(version.NewVersion(minNodeVersion))
+	if v.LessThan(min) {
+		return fmt.Errorf("node.js version %s or higher is required", minNodeVersion)
+	}
+
+	return nil
+}
+
+func checkDocker(ctx context.Context) error {
+	out, err := newCmd(ctx, "docker", "info", "--format", "json").Output()
+	if err != nil {
+		return fmt.Errorf("error executing the 'docker info' command: %w", err)
+	}
+
+	info := make(map[string]any)
+	err = json.Unmarshal(out, &info)
+	if err != nil {
+		return fmt.Errorf("error parsing the output of 'docker info': %w", err)
+	}
+
+	if sv, ok := info["ServerVersion"].(string); !ok || sv == "" {
+		return errors.New("error extracting the Docker server version")
+	}
+
+	if se, ok := info["ServerErrors"].([]string); ok && len(se) > 0 {
+		return fmt.Errorf("docker not available: found errors: %v", se)
+	}
+
+	return nil
+}
+
+func checkRillRepo(ctx context.Context) error {
+	_, err := os.Stat(".git")
+	if err != nil {
+		return fmt.Errorf("you must run `rill devtool` from the root of the rill repository")
+	}
+
+	out, err := newCmd(ctx, "git", "remote", "get-url", "origin").Output()
+	if err != nil {
+		return fmt.Errorf("error executing the 'git remote get-url origin' command: %w", err)
+	}
+
+	repo := strings.TrimSpace(string(out))
+	rillRepo := "https://github.com/rilldata/rill.git"
```

Comment:
Yeah I thought about that case, but decided to keep it simple for now, especially since this tool (with the `.env` sharing and all) is anyway only for internal use.